app/serializers/user_bookmark_serializer.rb

```diff
@@ -88,7 +89,15 @@ def closed
   end
 
   def highest_post_number
-    topic.highest_post_number
+    scope.is_staff? ? topic.highest_staff_post_number : topic.highest_post_number
+  end
+
+  def last_read_post_number
+    topic_user&.last_read_post_number
+  end
+
+  def topic_user
+    @topic_user ||= topic.topic_users.find { |tu| tu.user_id == scope.user.id }
```

Comment:
Just to confirm, `topic_users` has been preloaded right?

Reply:
Yes